encrypted-config-value/src/main/java/com/palantir/config/crypto/EncryptedValue.java

```diff
@@ -18,21 +18,32 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.common.io.BaseEncoding;
 import com.palantir.config.crypto.algorithm.Algorithm;
 import com.palantir.config.crypto.algorithm.Algorithms;
 import java.io.IOException;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class EncryptedValue {
 
     private static final String PREFIX = "enc:";
 
-    public abstract String encryptedValue();
+    @Nullable
+    public abstract String getAlgorithm();
```

Comment:
"AES" doesn't sufficiently encode the parameter space. The current algorithm is AES, but has a different IV length. Ideally we'd also be encoding the IV length, and we'd set the default for these missing parameters to be AES and the current IV length.